llvm_objects.py

```diff
@@ -244,31 +280,48 @@ class W_PhiInstruction(W_BaseInstruction):
     ''' Represents a wrapper for an LLVMValueRef that is a phi
         instruction. '''
 
+    _immutable_fields_ = ['l_instr', 'count_incoming', 'incoming_block[*]', 'incoming_value[*]']
+
     def __init__(self, l_instr, w_module):
         W_BaseInstruction.__init__(self, l_instr, w_module)
+        self.l_instr = l_instr
         self.count_incoming = llwrap.LLVMCountIncoming(l_instr)
-        self.incoming_block = []
-        self.incoming_value = []
+        self.incoming_block = None
+        self.incoming_value = self._get_incoming_value(l_instr)
+
+    def _get_incoming_value(self, l_instr):
```

Comment:
Can this only ever be called from `__init__`? If so, I suggest merging it into the `__init__` method, as it's clearly quite tightly coupled (e.g. the `[:]` immutable copy thing).